util/threadpool_imp.cc

```diff
@@ -110,6 +142,10 @@ struct ThreadPoolImpl::Impl {
 
  int total_threads_limit_;
  std::atomic_uint queue_len_;  // Queue length. Used for stats reporting
+ // Number of reserved threads by other tasks in bgthreads_
+ int reserved_threads_;
+ // Number of waiting threads (Maximum number of threads that can be reserved)
+ int num_waiting_;
```

Comment:
nit: num_waiting_threads_

(I generally prefer more explicit naming for variable so that people don't have to look at comment to understand its intention)

Reply:
Sure.